src/redprl/signature.sml

```diff
@@ -449,21 +449,24 @@ struct
           val (arguments', metactx) = elabDeclArguments arguments
           val (params', symctx, env) = elabDeclParams sign params
         in
-          E.wrap (pos, fn () => elabSrcRuleSpec (metactx, symctx, env) spec) >>= (fn (subgoalsSpec, seqjdg as hyps >> concl) =>
-            let
-              val tau = CJ.synthesis concl
-              val (params'', symctx', env') = 
-                Hyps.foldr
-                  (fn (x, jdg, (ps, ctx, env)) => 
-                    ((x, RedPrlSortData.HYP tau) :: ps, Tm.Sym.Ctx.insert ctx x (RedPrlSortData.HYP tau), NameEnv.insert env (Sym.toString x) x)) 
-                  (params', symctx, env)
-                  hyps
-            in
-              convertToAbt (metactx, symctx', env') script TAC 
-                >>= (fn scriptTm => elabRefine sign (seqjdg, scriptTm))
-                >>= checkProofState (pos, subgoalsSpec)
-                >>= (fn state => E.ret @@ EDEF {sourceOpid = opid, params = params'', arguments = arguments', sort = tau, spec = SOME seqjdg, state = state})
-            end)
+          E.wrap (pos, fn () => elabSrcRuleSpec (metactx, symctx, env) spec) >>=
+            (fn (subgoalsSpec, seqjdg as hyps >> concl) =>
+              let
+                val tau = CJ.synthesis concl
+                val (params'', symctx', env') =
+                  Hyps.foldr
+                    (fn (x, jdg, (ps, ctx, env)) =>
+                      ((x, RedPrlSortData.HYP tau) :: ps, Tm.Sym.Ctx.insert ctx x (RedPrlSortData.HYP tau), NameEnv.insert env (Sym.toString x) x))
+                    (params', symctx, env)
+                    hyps
+              in
+                convertToAbt (metactx, symctx', env') script TAC
+                  >>= (fn scriptTm => elabRefine sign (seqjdg, scriptTm))
+                  >>= checkProofState (pos, subgoalsSpec)
+                  >>= (fn state => E.ret @@ EDEF {sourceOpid = opid, params = params'', arguments = arguments', sort = tau, spec = SOME seqjdg, state = state})
+              end
+             | _ => raise Fail "Elaboration failed: no context in the derived rule")
```

Comment:
Let's leave the partial binding here rather than adding the wildcard case, as this is another invariant which is ensured elsewhere.